src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -1083,14 +1083,25 @@ protected virtual IEnumerable<SelectListItem> GetEnumSelectList([NotNull] ModelM
             }
 
             var selectList = new List<SelectListItem>();
-            foreach (var keyValuePair in metadata.EnumDisplayNamesAndValues)
+            var groupList = new Dictionary<string, SelectListGroup>();
+            foreach (var keyValuePair in metadata.EnumGroupedDisplayNamesAndValues)
             {
                 var selectListItem = new SelectListItem
                 {
-                    Text = keyValuePair.Key,
+                    Text = keyValuePair.Key.Item2,
                     Value = keyValuePair.Value,
                 };
 
+                if(string.IsNullOrEmpty(keyValuePair.Key.Item1) == false)
```

Comment:
- Match the style of other code in this repo: Space after `if`.
- Don't compare a non-`Nullable` value with `false`. Just `!string.IsNullOrEmpty(keyValuePair.Key.Item1)`.